### DISCUSSION TITLE
DPY-4011: the database or network closed the connection

### DISCUSSION THREAD

**Discussion by alexander-flouris** [oracle/python-oracledb#231] [Q&A] [answered]
Hi all,

I have written a small python script to execute a sql against my oracle database server version: 19.16.0.0.220719

```
import oracledb

un = ""
pw = ""

dsn = oracledb.makedsn("", "1521", sid="")
with oracledb.connect(user=un, password=pw, dsn=dsn, encoding="UTF-8") as connection:
    with connection.cursor() as cursor:
        sql = "SELECT COUNT(*) FROM ..."
        for r in cursor.execute(sql):
            print(r)
```
      
I do receive following error:

`site-packages\oracledb\errors.py", line 127, in _raise_err
    raise exc_type(_Error(message)) from cause
oracledb.exceptions.DatabaseError: DPY-4011: the database or network closed the connection
[WinError 10054] An existing connection was forcibly closed by the remote host`

On the database server I can see that a connection was established:

`19-SEP-2023 12:59:52 * (CONNECT_DATA=(SID=)(CID=(PROGRAM=...\python.exe)(HOST=...)(USER=...))(CONNECTION_ID=...==)) * (ADDRESS=(PROTOCOL=tcp)(HOST=...)(PORT=...)) * establish * ...* 0`

Anyone come across the same issue?

**Comment by anthony-tuininga** [comment, marked as answer]
Take a look at #170. I strongly suspect that you have NNE enabled and this is not supported by thin mode currently.

**Reply by alexander-flouris** [reply]
Hi Anthony, exactly what you said. It works for me now.
